sympy/printing/tests/test_tensorflow.py

```diff
@@ -0,0 +1,40 @@
+from sympy.printing.lambdarepr import TensorflowPrinter
+from sympy.printing.lambdarepr import _lambdify_tensorflow
+from sympy import (eye, symbols, MatrixSymbol, Symbol)
+
+from sympy.utilities.pytest import skip
+from sympy.external import import_module
+
+tf = import_module("tensorflow")
+
+def tensorflow_code(expr):
+    printer = TensorflowPrinter()
+    return printer.doprint(expr)
+
+
+def test_tensorflow_matrix():
+    if not tf:
+        skip("TensorFlow not installed")
+
+    assert tensorflow_code(eye(3)) == "tensorflow.Variable([[1, 0, 0], [0, 1, 0], [0, 0, 1]])"
+
+    M = MatrixSymbol("M", 3, 3)
+    N = MatrixSymbol("N", 3, 3)
+
+    expr = M
+    assert tensorflow_code(expr) == "M"
+
+    assert _lambdify_tensorflow((M,), expr) == """\
+def _lambdified(M_):
+    M = tensorflow.placeholder(tensorflow.float32, (3, 3))
+    feed_dict = {M: M_}
+    return tensorflow.InteractiveSession().run(M, feed_dict=feed_dict)
+"""
+    expr = M + N
+    assert _lambdify_tensorflow((M, N), expr) == """\
+def _lambdified(M_, N_):
+    M = tensorflow.placeholder(tensorflow.float32, (3, 3))
+    N = tensorflow.placeholder(tensorflow.float32, (3, 3))
+    feed_dict = {M: M_, N: N_}
+    return tensorflow.InteractiveSession().run(M + N, feed_dict=feed_dict)
+"""
```

Comment:
@asmeurer I use this trick to call tensorflow. Tensorflow requires the input data to be declared as placeholders, there was no way I could put this in a lambda as I need to define some local variables.

Furthermore the objects in `sympy.codegen` are of little use to me, as they all look to be optimized for C and Fortran code generation.

Probably this can be done in a better way, but I'd keep this trick as a temporary solution to concentrate on the code generation.

Reply:
That is fine. Even `lambdify` doesn't use a lambda anymore. 

Regarding codegen, what is missing? We would like for it to be as language agnostic as possible, but so far only C and Fortran have been in mind, so some of the abstractions may be wrong. CC @bjodah

Reply:
From what I've seen the function prototypes need to define the type already, which is the same for all variables of the function. This is a bit of a rigid concept, I think we should rather have the possibility of defining the type for each variable.

Anyway, there are other problems beyond functions that may arise, so I'll stick with this solution.